File: VB/XtraScheduler_XPO/XPObject.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.Xpo

Namespace XtraScheduler_XPO
	' XP object
	Public Class Task
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		' Appointment.AllDay
		Private _allDay As Boolean
		Public Property AllDay() As Boolean
			Get
				Return _allDay
			End Get
			Set(ByVal value As Boolean)
				SetPropertyValue("AllDay", _allDay, value)
			End Set
		End Property

		' Appointment.Description
		Private _description As String
		<Size(SizeAttribute.Unlimited)> _
		Public Property Description() As String
			Get
				Return _description
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Description", _description, value)
			End Set
		End Property

		' Appointment.End
		Private _finish As DateTime
		Public Property Finish() As DateTime
			Get
				Return _finish
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue("Finish", _finish, value)
			End Set
		End Property

		' Appointment.Label
		Private _label As Integer
		Public Property Label() As Integer
			Get
				Return _label
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("Label", _label, value)
			End Set
		End Property

		' Appointment.Location
		Private _location As String
		Public Property Location() As String
			Get
				Return _location
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Location", _location, value)
			End Set
		End Property

		' Appointment.RecurrenceInfo
		Private _recurrence As String
		<Size(SizeAttribute.Unlimited)> _
		Public Property Recurrence() As String
			Get
				Return _recurrence
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Recurrence", _recurrence, value)
			End Set
		End Property

		' Appointment.ReminderInfo
		Private _reminder As String
		<Size(SizeAttribute.Unlimited)> _
		Public Property Reminder() As String
			Get
				Return _reminder
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Reminder", _reminder, value)
			End Set
		End Property

		' Appointment.Start
		Private _start As DateTime
		Public Property Start() As DateTime
			Get
				Return _start
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue("Start", _start, value)
			End Set
		End Property

		' Appointment.Status
		Private _status As Integer
		Public Property Status() As Integer
			Get
				Return _status
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("Status", _status, value)
			End Set
		End Property

		' Appointment.Subject
		Private _subject As String
		Public Property Subject() As String
			Get
				Return _subject
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Subject", _subject, value)
			End Set
		End Property

		' Appointment.Type
		Private _appointmentType As Integer
		Public Property AppointmentType() As Integer
			Get
				Return _appointmentType
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("AppointmentType", _appointmentType, value)
			End Set
		End Property

		' Appointment.ResourceId
		Private _resourceId As Integer
		Public Property ResourceId() As Integer
			Get
				Return _resourceId
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("ResourceId", _resourceId, value)
			End Set
		End Property


			<Association()> _
			Public ReadOnly Property Resources() As XPCollection(Of XPResource)
			Get
				Return GetCollection(Of XPResource)("Resources")
			End Get
			End Property
	End Class

	' XP object
	Public Class XPResource
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		' Resource identifier
		Private _resourceId As Integer
		Public Property ResourceId() As Integer
			Get
				Return _resourceId
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("ResourceId", _resourceId, value)
			End Set
		End Property

		' Resource identifier
		Private _name As String
		Public Property Name() As String
			Get
				Return _name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _name, value)
			End Set
		End Property

		Private _color As Integer
		Public Property Color() As Integer
			Get
				Return _color
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("Color", _color, value)
			End Set
		End Property

		<Association()> _
		Public ReadOnly Property Tasks() As XPCollection(Of Task)
			Get
				Return GetCollection(Of Task)("Tasks")
			End Get
		End Property
	End Class



End Namespace

File: CS/XtraScheduler_XPO/XPObject.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.Xpo;

namespace XtraScheduler_XPO {
    // XP object
    public class Task : XPObject {
        public Task(Session session) : base(session) { }

        // Appointment.AllDay
        bool _allDay;
        public bool AllDay {
            get { return _allDay; }
            set { SetPropertyValue("AllDay", ref _allDay, value); }
        }

        // Appointment.Description
        string _description;
        [Size(SizeAttribute.Unlimited)]
        public string Description {
            get { return _description; }
            set { SetPropertyValue("Description", ref _description, value); }
        }

        // Appointment.End
        DateTime _finish;
        public DateTime Finish {
            get { return _finish; }
            set { SetPropertyValue("Finish", ref _finish, value); }
        }

        // Appointment.Label
        int _label;
        public int Label {
            get { return _label; }
            set { SetPropertyValue("Label", ref _label, value); }
        }

        // Appointment.Location
        string _location;
        public string Location {
            get { return _location; }
            set { SetPropertyValue("Location", ref _location, value); }
        }

        // Appointment.RecurrenceInfo
        string _recurrence;
        [Size(SizeAttribute.Unlimited)]
        public string Recurrence {
            get { return _recurrence; }
            set { SetPropertyValue("Recurrence", ref _recurrence, value); }
        }

        // Appointment.ReminderInfo
        string _reminder;
        [Size(SizeAttribute.Unlimited)]
        public string Reminder {
            get { return _reminder; }
            set { SetPropertyValue("Reminder", ref _reminder, value); }
        }

        // Appointment.Start
        DateTime _start;
        public DateTime Start {
            get { return _start; }
            set { SetPropertyValue("Start", ref _start, value); }
        }

        // Appointment.Status
        int _status;
        public int Status {
            get { return _status; }
            set { SetPropertyValue("Status", ref _status, value); }
        }

        // Appointment.Subject
        string _subject;
        public string Subject {
            get { return _subject; }
            set { SetPropertyValue("Subject", ref _subject, value); }
        }

        // Appointment.Type
        int _appointmentType;
        public int AppointmentType {
            get { return _appointmentType; }
            set { SetPropertyValue("AppointmentType", ref _appointmentType, value); }
        }

        // Appointment.ResourceId
        int _resourceId;
        public int ResourceId
        {
            get { return _resourceId; }
            set { SetPropertyValue("ResourceId", ref _resourceId, value); }
        }


            [Association()]
        public XPCollection<XPResource> Resources
        {
            get
            {
                return GetCollection<XPResource>("Resources");
            }
        }
    }

    // XP object
    public class XPResource : XPObject
    {
        public XPResource(Session session) : base(session) { }

        // Resource identifier
        int _resourceId;
        public int ResourceId
        {
            get { return _resourceId; }
            set { SetPropertyValue("ResourceId", ref _resourceId, value); }
        }

        // Resource identifier
        string _name;
        public string Name
        {
            get { return _name; }
            set { SetPropertyValue("Name", ref _name, value); }
        }

        int _color;
        public int Color
        {
            get { return _color;}
            set { SetPropertyValue("Color", ref _color, value); }
        }

        [Association()]
        public XPCollection<Task> Tasks
        {
            get
            {
                return GetCollection<Task>("Tasks");
            }
        }
    }



}